include/RAJA/index/ListSegment.hpp

```diff
@@ -148,14 +148,29 @@ class TypedListSegment
   ///
   /// \brief Construct list segment from given array with specified length.
   ///
-  /// By default the ctor performs deep copy of array elements.
+  /// By default the ctor on the host performs deep copy of array elements.
+  /// By default the ctor on the device does not perform a copy of array of elements. 
+  ///
+  RAJA_HOST_DEVICE TypedListSegment(const value_type* values,
+                                    Index_type length) 
+    : m_data{const_cast<value_type*>(values)}, m_size{length}, m_owned{Unowned}
+  {
+#if !defined(__CUDA_ARCH__)
```

Comment:
Should we make a `RAJA_DEVICE_CODE` define instead of using `__CUDA_ARCH__` directly? This seems like the kind of code change we would need for other devices too.